parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -306,33 +308,29 @@ private WriteBuilder createContextFunc(
       return this;
     }
 
+    // Utility method to get the column path
+    private String getParquetColumnPath(Map<Integer, String> fieldIdToParquetPath, String colPath) {
+      Types.NestedField fieldId = schema.findField(colPath);
+      if (fieldId == null) {
+        return null;
+      }
+
+      return fieldIdToParquetPath.get(fieldId.fieldId());
+    }
+
     private void setBloomFilterConfig(
         Context context,
-        MessageType parquetSchema,
+        Map<Integer, String> fieldIdToParquetPath,
         BiConsumer<String, Boolean> withBloomFilterEnabled,
         BiConsumer<String, Double> withBloomFilterFPP) {
 
-      Map<Integer, String> fieldIdToParquetPath =
-          parquetSchema.getColumns().stream()
-              .filter(col -> col.getPrimitiveType().getId() != null)
-              .collect(
-                  Collectors.toMap(
-                      col -> col.getPrimitiveType().getId().intValue(),
-                      col -> String.join(".", col.getPath())));
-
       context
           .columnBloomFilterEnabled()
           .forEach(
               (colPath, isEnabled) -> {
-                Types.NestedField fieldId = schema.findField(colPath);
-                if (fieldId == null) {
-                  LOG.warn("Skipping bloom filter config for missing field: {}", colPath);
```

Comment:
I think we dropped this warning after refactor, want to ensure this is expected

Reply:
Can add it back in getParquetColumnPath(), it will be a general warning which is not specific to bloom filter.

Reply:
If my understanding is correct, we dropped this message, but we start emitting `LOG.warn("Skipping bloom filter config for missing field: {}", colPath);` instead.

Reply:
I added the original behavior back by passing a configStr to getParquetColumnPath().